### ISSUE TITLE
RegexMatchError

### ISSUE THREAD

**Issue by emirpolatt** [pytube/pytube#382]
Hi guys. I was to installed pytube my computer yesterday. But after writing these simple codes:
### yt = YouTube("https://www.youtube.com/watch?v=VGo88XsIFLk")

I'm getting the following error:

`
  File "<ipython-input-1-8ec4d4992db4>", line 1, in <module>
    runfile('C:/Users/algan/.spyder-py3/temp.py', wdir='C:/Users/algan/.spyder-py3')

  File "C:\Users\algan\Anaconda3\lib\site-packages\spyder_kernels\customize\spydercustomize.py", line 786, in runfile
    execfile(filename, namespace)

  File "C:\Users\algan\Anaconda3\lib\site-packages\spyder_kernels\customize\spydercustomize.py", line 110, in execfile
    exec(compile(f.read(), filename, 'exec'), namespace)

  File "C:/Users/algan/.spyder-py3/temp.py", line 4, in <module>
    yt = YouTube("https://www.youtube.com/watch?v=5ICZ94ctqH8")

  File "C:\Users\algan\Anaconda3\lib\site-packages\pytube\__main__.py", line 88, in __init__
    self.prefetch_init()

  File "C:\Users\algan\Anaconda3\lib\site-packages\pytube\__main__.py", line 96, in prefetch_init
    self.prefetch()

  File "C:\Users\algan\Anaconda3\lib\site-packages\pytube\__main__.py", line 170, in prefetch
    age_restricted=self.age_restricted,

  File "C:\Users\algan\Anaconda3\lib\site-packages\pytube\extract.py", line 121, in video_info_url
    group=0,

  File "C:\Users\algan\Anaconda3\lib\site-packages\pytube\helpers.py", line 65, in regex_search
    .format(pattern=pattern),

RegexMatchError: regex pattern (\W[\'"]?t[\'"]?: ?[\'"](.+?)[\'"]) had zero matches`

Someone told me to change the variables in the cipher.py file, but I don't have cipher.py
How can ı solve ?

**Comment by stdedos** [comment]
Markdown pattern for block of code is:
````
```
````

not

````
`
````

This is a duplicate of #381

**Comment by ningjianli** [comment]
This is the problem I haven't solved！

**Comment by proy87** [comment]
Same for https://www.youtube.com/watch?v=67orVA_re_E&list=PLklMMMjn3IQk6BXvEOvbpvWDH_xbAQ_SS&index=18

**Comment by FearlessENT** [comment]
A new release has just been released a couple of days ago, it will probably have fixed it

**Comment by Jan-V2** [comment]
i have the latest version pulled straight from github and i have the exact same problem.
however i think i've figured that the problem is in the video_info_url function in extract .py:

what this function does is it extracts a bunch of variables from html and uses that to fetch info about the video from an api endpoint. the problem is in line 119-122 (in extract.py).

```python
t = regex_search(
            r'\W[\'"]?t[\'"]?: ?[\'"](.+?)[\'"]', watch_html,
            group=0,
        )
```
it searches the watch_html for the t variable but because it can't find a match (presumably because youtube changed something about their html), it throws an exception.

there is however a temporary fix for this, and that is to go line 170 in __main__.py , and change 
```python
            age_restricted=self.age_restricted,
```
to
```python
            age_restricted=True,
```
by doing this you pretend like every video is age restricted, which means that in the video_info_url function the query url will be constructed in a different way that isn't broken (by searching the embed_html rather than the search_html). this will allow you to once again download videos and view metadata. however this will mean that you'll have less metadata available then you would otherwise but depending on what you're trying to do that might not be a problem for you.

hope this was helpful.

**Comment by princelySid** [comment]
Issue seems fixed with pytube 9.5
